packages/httpsnippet-client-api/src/index.js

```diff
@@ -116,17 +116,33 @@ module.exports = function (source, options) {
 
   let metadata = {};
   if (Object.keys(source.queryObj).length) {
-    const queryParams = source.queryObj;
-
-    Object.keys(queryParams).forEach(param => {
+    Object.keys(source.queryObj).forEach(param => {
       if (authSources.query.includes(param)) {
-        authData.push(buildAuthSnippet(queryParams[param]));
+        authData.push(buildAuthSnippet(source.queryObj[param]));
 
-        delete queryParams[param];
+        // If this query param is part of an auth source then we don't want it doubled up in the
+        // snippet.
+        return;
       }
+
+      metadata[param] = source.queryObj[param];
     });
+  }
+
+  if (Object.keys(source.cookiesObj).length) {
```

Comment:
What is this `length` check preventing? Wouldn't the `forEach` essentially be a no-op on an empty array?

Reply:
Great point.